src/production.cpp

```diff
@@ -57,16 +59,10 @@ static void readTagsAndRunConveyor(TaskContext *ctx) {
           sniprintf(two, sizeof(two), "%02x", buffer[i]);
           tag += two;
         }
+        memcpy(values.production.tag, tag.c_str(), tag.length());
+        values.production.isNewTagPresent = true;
```

Comment:
Be careful when sharing values between threads/tasks.
Based on a quick look, it seems  possible for the Dolibarr task to read a tag that is partially initialized.
That may work currently, but concurrency issues are the kind that happen 1/1000 times (usually in live demos).

Non exhaustive guide to "atomics" in FreeRTOS
https://www.freertos.org/Documentation/02-Kernel/04-API-references/04-RTOS-kernel-control/01-taskENTER_CRITICAL_taskEXIT_CRITICAL

Reply:
Done
See commit 4616291ea0522a99f79aa97b6ed2c7299c892711